src/litdata/streaming/resolver.py

```diff
@@ -218,28 +219,27 @@ def _assert_dir_is_empty(output_dir: Dir, append: bool = False, overwrite: bool
 
     obj = parse.urlparse(output_dir.url)
 
-    if obj.scheme != "s3":
-        raise ValueError(f"The provided folder should start with s3://. Found {output_dir.url}.")
-
-    s3 = boto3.client("s3")
+    if obj.scheme not in _SUPPORTED_CLOUD_PROVIDERS:
```

Comment:
Looking good so far. Just a small note: we should also add a test to ensure that this error is triggered when unsupported cloud provider paths are passed.

I think currently some other error is taking precedence , may be I think if we could check in the `_resolve_dir` as well

Reply:
interesting. For an unsupported cloud, it will use `return Dir(path=dir_path_absolute, url=None)`

Let's say if user asks for `dropbox://some_path`, resolve_dir will consider it to be a local_path and not cloud.

This was the original behaviour. Should we do something or leave it as it is? What do you think!

Reply:
I think we can check if the obj scheme is provided in URLs with supported providers.

Reply:
Done!